static/src/js/components/PermissionForm/PermissionForm.jsx

```diff
@@ -0,0 +1,647 @@
+    skip: conceptId === 'new',
+    variables: {
+      conceptId
+    }
+  })
+
+  useEffect(() => {
+    if (conceptId === 'new') {
+      setDraft({})
+      setOriginalDraft({})
+    }
+  }, [conceptId])
+
+  const showGranuleFields = (formData) => {
+    if (formData.accessPermission?.granule) {
+      const newUiSchema = {
+        ...collectionPermissionUiSchema,
+        accessConstraintFilter: {
+          ...collectionPermissionUiSchema.accessConstraintFilter,
+          granuleAccessConstraint: {
+            ...collectionPermissionUiSchema.accessConstraintFilter.granuleAccessConstraint,
+            'ui:disabled': false
+          }
+        },
+        temporalConstraintFilter: {
+          ...collectionPermissionUiSchema.temporalConstraintFilter,
+          granuleTemporalConstraint: {
+            ...collectionPermissionUiSchema.temporalConstraintFilter.granuleTemporalConstraint,
+            'ui:disabled': false
+          }
+        }
+      }
+      setUiSchema(newUiSchema)
+    } else {
+      const newUiSchema = {
+        ...collectionPermissionUiSchema,
+        accessConstraintFilter: {
+          ...collectionPermissionUiSchema.accessConstraintFilter,
+          granuleAccessConstraint: {
+            ...collectionPermissionUiSchema.accessConstraintFilter.granuleAccessConstraint,
+            'ui:disabled': true
+          }
+        }
+      }
+      setUiSchema(newUiSchema)
+    }
+  }
+
+  useEffect(() => {
+    if (data) {
+      const { acl } = data
+      const {
+        name,
+        catalogItemIdentity,
+        groups,
+        collections
+      } = acl
+
+      const { items } = collections || {}
+
+      const selectedCollections = items?.map((item) => {
+        const {
+          conceptId: collectionConceptId,
+          directDistributionInformation,
+          provider,
+          shortName,
+          title
+        } = item
+
+        return {
+          conceptId: collectionConceptId,
+          directDistributionInformation,
+          entryTitle: title,
+          provider,
+          shortName
+        }
+      })
+
+      const searchAndOrderGroupPermission = []
+      const searchPermission = []
+
+      groups.items?.forEach((item) => {
+        const {
+          id,
+          name: groupName,
+          permissions,
+          userType,
+          tag
+        } = item
+
+        const groupObj = userType
+          ? {
+            value: `all-${userType}-user`,
+            label: `All ${userType} user`,
+            isDisabled: true
+          }
+          : {
+            value: id,
+            label: groupName,
+            provider: tag,
+            isDisabled: true
+          }
+
+        if (permissions.length === 2) {
+          searchAndOrderGroupPermission.push(groupObj)
+        } else {
+          searchPermission.push(groupObj)
+        }
+      })
+
+      const {
+        collectionApplicable,
+        granuleApplicable,
+        collectionIdentifier,
```

Comment:
Sorting these lines makes them easier to read. Any time you are dealing with an object, destructuring or writing, sort the lines to make it easier to read and maintain